File: src/main/java/org/nanowrimo/ws/WordCountHandler.java
package org.nanowrimo.ws;

import org.xml.sax.ContentHandler;
import org.xml.sax.DTDHandler;
import org.xml.sax.EntityResolver;
import org.xml.sax.ErrorHandler;
import org.xml.sax.helpers.DefaultHandler;


/**
 * @author Paul S. Hawke (paul.hawke@gmail.com)
 * On: Sep 17, 2007 at 4:40:43 PM
 */
abstract class WordCountHandler<T extends WordCountSummary> extends DefaultHandler implements EntityResolver, DTDHandler, ContentHandler, ErrorHandler, ServiceApiConstants {
    abstract T getSummary();
}

File: src/main/java/org/nanowrimo/ws/WordCountHandler.kt
package org.nanowrimo.ws

import org.xml.sax.ContentHandler
import org.xml.sax.DTDHandler
import org.xml.sax.EntityResolver
import org.xml.sax.ErrorHandler
import org.xml.sax.helpers.DefaultHandler

// user id
const val USER_NAME = "uname"
const val USER_ID = "uid"
const val USER_WORD_COUNT = "user_wordcount"
const val USER_WINNER = "winner"
const val USER_ERROR = "error"

// region
const val REGION_NAME = "rname"
const val REGION_ID = "rid"
const val REGION_WORD_COUNT = "region_wordcount"
const val REGION_MIN = "min"
const val REGION_MAX = "max"
const val REGION_AVERAGE = "avg"
const val REGION_STDDEV = "stddev"
const val REGION_COUNT = "count"

// shared
const val WORD_COUNT_ENTRY = "wcentry"
const val WORD_COUNT_ENTRY_DATE = "wcdate"
const val WORD_COUNT_RANK = "wcrank"
const val WORD_COUNT = "wc"
const val WORD_COUNT_GOAL = "goal"
const val WORDCOUNTS_CONTAINER = "wordcounts"

internal abstract class WordCountHandler<out T : WordCountSummary> : DefaultHandler(), EntityResolver, DTDHandler, ContentHandler, ErrorHandler {
    internal abstract val summary: T
}
